maintenance/setupStore.php

```diff
@@ -111,6 +113,24 @@ public function execute() {
 			$messageReporter->registerReporterCallback( array( $this, 'reportMessage' ) );
 		}
 
+		// #2963 Use the DB handler from the Maintenance script which may access
+		// the `$wgDBadminuser` instead of the regular `$wgDBuser`
+		$callbackConnectionProvider = new CallbackConnectionProvider( function() {
+				return $this->getDB( DB_MASTER );
+			}
+		);
+
+		$connectionManager = new ConnectionManager();
+
+		$connectionManager->registerConnectionProvider(
+			DB_MASTER,
+			$callbackConnectionProvider
+		);
+
+		$store->setConnectionManager(
+			$connectionManager
+		);
+
```

Comment:
Looks like this should be a single function. No need to have these two new variables in scope longer, or to access the ones already in scope. Plus this function is already very long, with lots of random details (thus mixed levels of abstractions) in it.